src/main/java/sirius/biz/jobs/params/FileParameter.java

```diff
@@ -23,38 +24,93 @@
 
 /**
  * Permits to select a directory in the {@link VirtualFileSystem}.
- *
- * @param <P> the effective parameter type for fluent method calls
  */
-public abstract class BaseFileParameter<P extends ParameterBuilder<VirtualFile, P>> extends
-                                                                                    ParameterBuilder<VirtualFile, P> {
+public class FileParameter extends ParameterBuilder<VirtualFile, FileParameter> {
 
     @Part
-    protected static VirtualFileSystem vfs;
+    private static VirtualFileSystem vfs;
 
-    protected String basePath;
-    protected List<String> acceptedExtensions;
+    private String basePath;
+    private List<String> acceptedExtensions;
+    private boolean allowFile = true;
+    private boolean allowDirectory = true;
 
     /**
      * Creates a new parameter with the given name and label.
      *
      * @param name  the name of the parameter
      * @param label the label of the parameter, which will be {@link NLS#smartGet(String) auto translated}
      */
-    protected BaseFileParameter(String name, String label) {
+    public FileParameter(String name, String label) {
         super(name, label);
     }
 
+    /**
+     * Allow only files.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    public FileParameter filesOnly() {
+        this.allowFile = true;
+        this.allowDirectory = false;
+        return self();
+    }
+
+    /**
+     * Allow only directories.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    public FileParameter directoriesOnly() {
+        this.allowFile = false;
+        this.allowDirectory = true;
+        return self();
+    }
+
+    /**
+     * Allow both files and directories.
+     * <p>
+     * This is the default behavior, but you are encouraged to specify it anyway, as it enhances readability.
+     *
+     * @return the parameter itself for fluent method calls
+     */
+    public FileParameter filesAndDirectories() {
+        this.allowFile = true;
+        this.allowDirectory = true;
+        return self();
+    }
+
+    public boolean isFilesOnly() {
+        return allowFile && !allowDirectory;
+    }
+
+    /**
+     * Whether files are allowed as parameter.
+     *
+     * @return true if files are allowed
+     */
+    public boolean allowFile() {
```

Comment:
Optional: Rename to `allowsFile`. The wording `allowFile` implies it is actually allowing files and not simply checking what the current value is.